packages/popover/src/Popover/Popover.tsx

```diff
@@ -160,14 +175,37 @@ export const Popover = forwardRef<HTMLDivElement, PopoverComponentProps>(
       return children;
     };
 
+    useEffect(() => {
+      if (renderMode !== RenderMode.TopLayer || !onToggle) {
+        return;
+      }
+
+      // @ts-expect-error - `toggle` event not supported pre-typescript v5
+      refs.floating.current?.addEventListener('toggle', onToggle);
+      return () =>
+        // @ts-expect-error - `toggle` event not supported pre-typescript v5
+        refs.floating.current?.removeEventListener('toggle', onToggle);
+    }, [onToggle, renderMode]);
```

Comment:
might need a eslint-ignore here to stop the linter complaining about `refs.floating`